### PR TITLE
Adding classes for parallel phenology

### PR DESCRIPTION
**Pull request by HamishBrownPFR**
working on #8551

I have needed to do parts of the phenology in parllell for DEROPAPY.  The problem is that we use phenology to mark key events for the canopy and for biomass partitioning.  However, DEROPAPY represent a broad range of crops and key events (e.g start reproductive growth vs full canopy) do not alway happen in the same sequence.

The files in this PR represent an intum solution where I use the main phenology model to represent events in the canopy (start growth, full canopy, start senescence, canopy death) and I have added a concept of parallel phases that can start at a certian point and run through a seperate sequence to represent partitioning events (vegetative, flowering, reproductive growth).

This has got things working well enough to progress with the DEROPAPY prototype and I need to get this locked in to complete the first phase of development.

A better long-term solution would be to change Phenology a crop can have two (or more) phenology models driving different aspects of phenology.  I did try to do this but it got complicated quickly.  Now I have something that does what I want functionally, I will come back to it and refactor into something more tidy.

### CODE REVIEW

**Comment by HamishBrownPFR** [comment]
@hol353 @par456 @ric394 can someone please review and merge when happy.  I am not sure why the stats have improved for the clover tests but the improvements are very small.

File: Models/PMF/Phenology/Phases/GotoPhase.cs [hunk 1]
```diff
@@ -49,23 +50,37 @@ public string End
         [Display(Type = DisplayType.CropPhaseName)]
         public string PhaseNameToGoto { get; set; }
 
+        /// <summary>
+        /// The type of biomass removal event
+        /// </summary>
+        [Description("Type of biomass removal.  This triggers events OnCutting, OnGrazing etc")]
+        public BiomassRemovalType RemovalType
```
Comment (hol353) [inline Models/PMF/Phenology/Phases/GotoPhase.cs:57]: Why is biomass removal stuff being put into a GotoPhase function?

File: Models/PMF/Phenology/Phases/GrazeAndRewindPhase.cs [hunk 2]
```diff
@@ -35,7 +35,7 @@ public string End
         {
             get
             {
-                return phenology?.FindChild<IPhase>(PhaseNameToGoto)?.Start;
+                return "Rewind phase has no end";//phenology?.FindChild<IPhase>(PhaseNameToGoto)?.Start;
```
Comment (hol353) [inline Models/PMF/Phenology/Phases/GrazeAndRewindPhase.cs:38]: Remove commented code.

File: Models/PMF/Phenology/Phases/GotoPhase.cs (hunk 1 above)
Comment (HamishBrownPFR) [reply Models/PMF/Phenology/Phases/GotoPhase.cs:57]: The go to phase broadcasts a PhenologyDefoliate event that has an arguement of RemovalType.
The BiomassRemovalEvent subscribes to this event and makes the appropriate biomass removal happen when ever the goToPhase is reached.  This mimics what the GrazeAndRewind phase does and means we can deprocate that phase as it is now redundant.  I still need to replace the GrazeAndRewind phase in SPRUM with the more generic GoToPhase before I can deprocate the class though.

File: Models/PMF/Phenology/Phases/GrazeAndRewindPhase.cs (hunk 2 above)
Comment (HamishBrownPFR) [reply Models/PMF/Phenology/Phases/GrazeAndRewindPhase.cs:38]: That class is marked for deprecation.  I will remove the commented code in my next pull request to save Jenkens the job of running the tests again

**Comment by HamishBrownPFR** [comment]
@hol353 are these Reponses OK.

**Pull request merged**